src/containers/Auth/SignUp/SignUp.js

```diff
@@ -0,0 +1,153 @@
+import React, { Component } from "react";
+import { Form, Button, Grid, Header, Radio } from "semantic-ui-react";
+import { Link } from "react-router-dom";
+import { registerUser } from "../../../redux/actions/userAction";
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import formHoc from '../../../hoc/formHoc';
+
+const INITIAL_STATE = {
+  firstname: "",
+  lastname: "",
+  password: "",
+  email: "",
+  usertype: "",
+  error: null,
+};
+
+export class SignUpFormBase extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...INITIAL_STATE };
+  }
+ emailValidation(){
+    const regex = /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+    if(!this.state.email || regex.test(this.state.email) === false){
+        this.setState({
+            error: "Email is not valid"
+        });
+        return false;
+    }
```

Comment:
Since this is repetition of code, you can consider validation js as separate file and use it as common re-usable validation file for all kind of form validation.